test/widgets/ride_detail_page_test.dart

```diff
@@ -0,0 +1,554 @@
+    });
+
+    group('when ride is pending', () {
+      setUp(() {
+        ride = RideFactory().generateFake(status: RideStatus.pending, drive: NullableParameter(drive));
+
+        when(processor.processUrl(any)).thenReturn(jsonEncode(ride.toJsonForApi()));
+      });
+
+      testWidgets('it shows a banner and the withdraw button, but no riders', (WidgetTester tester) async {
+        await pumpMaterial(tester, RideDetailPage.fromRide(ride));
+        await tester.pump();
+
+        expect(find.byKey(const Key('rideRequestedBanner')), findsOneWidget);
+        expect(find.byKey(const Key('withdrawRideButton')), findsOneWidget);
+        expect(find.byType(ProfileWrapList), findsNothing);
+      });
+    });
+
+    group('when ride is approved', () {
+      testWidgets('it shows the riders', (WidgetTester tester) async {
+        await pumpMaterial(tester, RideDetailPage.fromRide(ride));
+        await tester.pump();
+
+        expect(find.byType(ProfileWrapList), findsOneWidget);
+      });
+
+      testWidgets('it shows the cancel button when ride is ongoing', (WidgetTester tester) async {
+        await pumpMaterial(tester, RideDetailPage.fromRide(ride));
+        await tester.pump();
+
+        expect(find.byKey(const Key('cancelRideButton')), findsOneWidget);
+      });
+
+      testWidgets('it shows the rating button when ride is finished', (WidgetTester tester) async {
+        ride = RideFactory().generateFake(
+          status: RideStatus.approved,
+          endTime: DateTime.now().subtract(const Duration(hours: 1)),
+          drive: NullableParameter(drive),
+        );
+
+        when(processor.processUrl(any)).thenReturn(jsonEncode(ride.toJsonForApi()));
+
+        await pumpMaterial(tester, RideDetailPage.fromRide(ride));
+        await tester.pump();
+
+        expect(find.byKey(const Key('rateDriverButton')), findsOneWidget);
+      });
+    });
+
+    group('when ride is rejected', () {
+      setUp(() {
+        ride = RideFactory().generateFake(status: RideStatus.rejected, drive: NullableParameter(drive));
+
+        when(processor.processUrl(any)).thenReturn(jsonEncode(ride.toJsonForApi()));
+      });
+
+      testWidgets('it shows the hide button and a banner, but no riders', (WidgetTester tester) async {
+        await pumpMaterial(tester, RideDetailPage.fromRide(ride));
+        await tester.pump();
+
+        expect(find.byKey(const Key('rideRejectedBanner')), findsOneWidget);
+        expect(find.byKey(const Key('hideRideButton')), findsOneWidget);
+        expect(find.byType(ProfileWrapList), findsNothing);
+      });
+    });
+
+    group('when ride is cancelled by driver', () {
+      setUp(() {
+        ride = RideFactory().generateFake(status: RideStatus.cancelledByDriver, drive: NullableParameter(drive));
+
+        when(processor.processUrl(any)).thenReturn(jsonEncode(ride.toJsonForApi()));
+      });
+
+      testWidgets('it shows the riders, a hide button and a banner', (WidgetTester tester) async {
```

Comment:
why are we showing the riders here but not when the ride is cancelledbyRider?

Reply:
well if its cancelledByDriver, then the state is basically frozen - no one can request or withdraw etc. So it's fine if show all the other riders (only overlapping of course) that were equally screwed by the cancellation.

But if it's cancelledByRider, then you are not a part of the drive anymore, so you should not see any riders that are added after you cancelled for example

Reply:
that makes sense thanks